src/Core/Managed/Shared/TelemetryClient.cs

```diff
@@ -17,14 +17,19 @@
     /// <summary>
     /// Send events, metrics and other telemetry to the Application Insights service.
     /// </summary>
-    public sealed class TelemetryClient
+    public sealed class TelemetryClient : IDisposable
```

Comment:
I agree 100%.
TelemetryChannel should not be creating or managing the channel, but currently it does and we will potentially break customers if we start throwing exceptions now (not that I'm opposed to that if others are okay with it).
I will happily change it if Sergey agrees because that will simplify and remove a lot of this problem and fix #364.
I 100% want to throw exceptions instead of trying to work around user mistakes, what it does right now is just open to mistakes and/or abuse.